lib/HelloSignResource.js

```diff
@@ -331,11 +331,19 @@ HelloSignResource.prototype = {
             // submit as form
             var form = new FormData();
             for(var i=0; i<data.files.length;i++) {
-                form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                if( typeof data.files[i] === 'string' ) {
+                    form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                }
+                else if( data.files[i] instanceof Buffer ) {
+                    form.append("file[" + i + "]", data.files[i], {filename: 'signature.pdf'});
```

Comment:
I'm a little hesitant about this default filename of `signature.pdf`.  Is there an advantage in allowing people to just specify the buffers without a filename?  If we allow buffers I'm a little more inclined to _require_ that the object form is passed with the filename included (like below).

Reply:
A file name must be provided for the buffer to work within the FormData class. So either an exception must be thrown or a default name used. I wanted to minimize impact to anyone's code & since the name is unused by HelloSign and only used for the purposes of getting the buffer support that's why I just hard coded it with a sensible name.

If for some reason this was to be used it would make sense to extract the file name from the file path (for non-buffer case) as well as allow the user potentially to specify a name in the case of a buffer for uniformity. For practical purposes of getting in buffer support while maximizing backwards compatibility I chose to use a static name.

Reply:
Since it's not used, could it be something more descriptive to that effect?
 e.g., `placeholder_only_filename_not_used.pdf`?